.github/workflows/provider-publish-service-artifacts.yml

```diff
@@ -0,0 +1,100 @@
+name: Provider Publish Service Artifacts
+
+on:
+  workflow_call:
+    inputs:
+      subpackages:
+        description: 'Subpackages to be built individually (e.g. monolith config ec2)'
+        required: true
+        type: string
+    secrets:
+      UPBOUND_MARKETPLACE_PUSH_ROBOT_USR:
+        required: true
+      UPBOUND_MARKETPLACE_PUSH_ROBOT_PSW:
+        required: true
+
+env:
+  # Common versions
+  GO_VERSION: '1.19'
+  GOLANGCI_VERSION: 'v1.50.0'
+  DOCKER_BUILDX_VERSION: 'v0.8.2'
+
+  # Common users. We can't run a step 'if secrets.XXX != ""' but we can run a
+  # step 'if env.XXX' != ""', so we copy these to succinctly test whether
+  # credentials have been provided before trying to run steps that need them.
+  UPBOUND_MARKETPLACE_PUSH_ROBOT_USR: ${{ secrets.UPBOUND_MARKETPLACE_PUSH_ROBOT_USR }}
+  
+jobs:  
+  publish-service-artifacts:
+    if: ${{ github.actor == 'OWNER' || github.actor == 'turkenf' || contains(github.event.repository.collaborators, github.actor) }}
+    runs-on: ubuntu-22.04
+
+    steps:
```

Comment:
Not with this initial PR but as a next step, we should consider parallelizing [using a matrix strategy](https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs#using-a-matrix-strategy). 

```yaml
jobs:
  package_matrix:
    strategy:
      matrix:
        subpackage: [config, ec2, rds] # consider calculating this from sub packages input.
        arch: [amd64, arm64]
```

Reply:
I think we had better implement this @turkenf and start experimenting in this way.